articles/openshift/oshift-how-build-app-private-repo.md

```diff
@@ -0,0 +1,119 @@
+---
+title: How to build an OpenShift application from a private Github repo | UKCloud Ltd
+description: Shows how to build an OpenShift application where the source code is in a "private" github repository secured with an SSH Key Pair.
+services: openshift
+author: Andy Watkins
+
+toc_rootlink: How To
+toc_sub1:
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Build an OpenShift app from a private Github repo
+toc_fullpath: How To/oshift-how-build-app-private-repo.md
+toc_mdlink: oshift-how-build-app-private-repo.md
+---
+
+# How to build an OpenShift application from a private Github repo
+
+This guide assumes you have access to the command-line `oc` client, and have
+logged in to your openshift instance with `oc login`
+
+With a private Github repo, you want to ensure it stays hidden from prying
+eyes, but you also want your Openshift app to be able to build from the repo.
+The way around this is to use a 
+[Deploy key](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys).
+
+
+### Steps:
+**First**
+create an ssh key-pair (don't use your regular one, and make sure
+you don't overwrite your regular one either!!)
+ 
+```bash
+ssh-keygen -t rsa -b 4096 -c "jbloggs@mycompany.com" -f my_github_deploy_key
+```
+This will generate both the private and public key files. The public one will
+have a ".pub" suffix.
+
+
+
+**Second**, add the public key to the repo as a **Deploy Key** (instructions
+with screen-shots on 
+[Github](https://developer.github.com/v3/guides/managing-deploy-keys/#deploy-keys))
+
+
+
+**Third**, add the private key to your OpenShift instance. 
+Note that the word "mygithubsecret" is the name of your secret, not a password.
+ 
+```bash
+oc secrets new-sshauth mygithubsecret --ssh-privatekey=./my_github_deploy_key
+```
+
+
+
+**Fourth**, attempt a build, which will fail, (but will enable you to add the
```

Comment:
4. Attempt a build...
You don't need the parentheses.
Make sure each sentence ends with a full stop.